File: check_readiness.py
import requests
import time

base = "https://lid-webster-impaired-cut.trycloudflare.com/api/v1"
email = f"test_{int(time.time())}@gmail.com"

# Register
res = requests.post(f"{base}/auth/register", json={"email": email, "password": "Password123!", "full_name": "Test User"})
token = res.json()["access_token"]
headers = {"Authorization": f"Bearer {token}"}

# Workspace
ws = requests.post(f"{base}/workspaces", headers=headers, json={"name": "W1"}).json()
ws_id = ws["id"]

# Project
proj = requests.post(f"{base}/projects?workspace_id={ws_id}", headers=headers, json={"name": "P1", "description": ""}).json()
p_id = proj["id"]

print("Project:", p_id)

# Readiness initial
read_init = requests.get(f"{base}/projects/{p_id}/pipeline/readiness", headers=headers).json()
print("Initial:", read_init)

# Upload FASTA
with open("e2e/fixtures/protein.fasta", "rb") as f:
    u1 = requests.post(f"{base}/projects/{p_id}/files/upload", headers=headers, files={"file": f}).json()
print("Fasta file:", u1)
requests.patch(f"{base}/projects/{p_id}/inputs/files", headers=headers, json={"fasta_file_id": u1["file_id"]})

# Upload PDB
with open("e2e/fixtures/protein.pdb", "rb") as f:
    u2 = requests.post(f"{base}/projects/{p_id}/files/upload", headers=headers, files={"file": f}).json()
print("PDB file:", u2)
requests.patch(f"{base}/projects/{p_id}/inputs/files", headers=headers, json={"protein_structure_file_id": u2["file_id"]})

# Final readiness
read_final = requests.get(f"{base}/projects/{p_id}/pipeline/readiness", headers=headers).json()
print("Final:", read_final)

File: check_readiness.js
const fs = require("fs");
const path = require("path");
const fetch = require("node-fetch");
const FormData = require("form-data");

async function test() {
  const base = "https://lid-webster-impaired-cut.trycloudflare.com/api/v1";
  const email = "test_" + Date.now() + "@gmail.com";
  
  // Register
  const regRes = await fetch(base + "/auth/register", {
    method: "POST", headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ email, password: "Password123!", full_name: "Test User" })
  });
  const auth = await regRes.json();
  const token = auth.access_token;
  const headers = { "Authorization": `Bearer ${token}`, "Content-Type": "application/json" };
  
  // Workspace
  const wsRes = await fetch(base + "/workspaces", { method: "POST", headers, body: JSON.stringify({ name: "W1" }) });
  const ws = await wsRes.json();
  
  // Project
  const pRes = await fetch(base + "/projects?workspace_id=" + ws.id, { method: "POST", headers, body: JSON.stringify({ name: "P1", description: "" }) });
  const proj = await pRes.json();
  const projectId = proj.id;
  
  console.log("Project:", projectId);
  
  // Check Readiness
  let readRes = await fetch(base + "/projects/" + projectId + "/pipeline/readiness", { headers });
  console.log("Initial:", await readRes.text());
  
  // Upload FASTA
  const fData1 = new FormData();
  fData1.append("file", fs.createReadStream(path.resolve(__dirname, "e2e/fixtures/protein.fasta")));
  const u1 = await fetch(base + "/projects/" + projectId + "/files/upload", { method: "POST", headers: { "Authorization": `Bearer ${token}` }, body: fData1 });
  const file1 = await u1.json();
  console.log("File1:", file1);
  
  await fetch(base + "/projects/" + projectId + "/inputs/files", { method: "PATCH", headers, body: JSON.stringify({ fasta_file_id: file1.file_id }) });
  
  // Upload PDB
  const fData2 = new FormData();
  fData2.append("file", fs.createReadStream(path.resolve(__dirname, "e2e/fixtures/protein.pdb")));
  const u2 = await fetch(base + "/projects/" + projectId + "/files/upload", { method: "POST", headers: { "Authorization": `Bearer ${token}` }, body: fData2 });
  const file2 = await u2.json();
  console.log("File2:", file2);
  
  await fetch(base + "/projects/" + projectId + "/inputs/files", { method: "PATCH", headers, body: JSON.stringify({ protein_structure_file_id: file2.file_id }) });
  
  // Check Readiness again
  readRes = await fetch(base + "/projects/" + projectId + "/pipeline/readiness", { headers });
  console.log("Final:", await readRes.text());
}
test().catch(console.error);
